src/evaluate/evaluator/text2text_generation.py

```diff
@@ -0,0 +1,269 @@
+    >>> data = load_dataset("dailymail_cnn", "3.0.0", split="test[:40]")
+    >>> results = task_evaluator.compute(
+    >>>     model_or_pipeline="facebook/bart-large-cnn",
+    >>>     data=data,
+    >>>     label_column="article",
+    >>>     label_column="highlights",
+    >>>     metric="rouge",
+    >>> )
+    ```
+"""
+
+TASK_DOCUMENTATION_SUMMARIZATION = r"""
+    Examples:
+    ```python
+    >>> from evaluate import evaluator
+    >>> from datasets import load_dataset
+    >>> task_evaluator = evaluator("summarization")
+    >>> data = load_dataset("dailymail_cnn", "3.0.0", split="test[:40]")
+    >>> results = task_evaluator.compute(
+    >>>     model_or_pipeline="facebook/bart-large-cnn",
+    >>>     data=data,
+    >>>     label_column="article",
+    >>>     label_column="highlights",
+    >>> )
+    ```
+"""
+
+TASK_DOCUMENTATION_TRANSLATION = r"""
+    Examples:
+    ```python
+    >>> from evaluate import evaluator
+    >>> from datasets import load_dataset
+    >>> task_evaluator = evaluator("translation")
+    >>> data = load_dataset("wmt19", "fr-de", split="test[:40]")
+    >>> data = data.map(lambda x: {"text": x["translation"]["de"], "label": x["translation"]["fr"]})
+    >>> results = task_evaluator.compute(
+    >>>     model_or_pipeline="Helsinki-NLP/opus-mt-de-fr",
+    >>>     data=data,
+    >>> )
+    ```
+"""
+
+
+class Text2TextGenerationEvaluator(Evaluator):
+    """
+    Text2Text generation evaluator.
+    This Text2Text generation evaluator can currently be loaded from [`evaluator`] using the default task name
+    `text2text`.
+    Methods in this class assume a data format compatible with the [`Text2TextGenerationPileine`].
+    """
+
+    PREDICTION_PREFIX = "generated"
+    PIPELINE_KWARGS = {"truncation": True}
+
+    def __init__(self, task="text2text-generation", default_metric_name=None):
+        super().__init__(task, default_metric_name=default_metric_name)
+
+    def predictions_processor(self, predictions, label_mapping):
+        return {"predictions": [pred[f"{self.PREDICTION_PREFIX}_text"] for pred in predictions]}
+
+    @add_start_docstrings(EVALUTOR_COMPUTE_START_DOCSTRING)
+    @add_end_docstrings(EVALUATOR_COMPUTE_RETURN_DOCSTRING, TASK_DOCUMENTATION_TEXT2TEXT)
+    def compute(
+        self,
+        model_or_pipeline: Union[
+            str, "Pipeline", Callable, "PreTrainedModel", "TFPreTrainedModel"  # noqa: F821
+        ] = None,
+        data: Union[str, Dataset] = None,
+        metric: Union[str, EvaluationModule] = None,
+        tokenizer: Optional[Union[str, "PreTrainedTokenizer"]] = None,  # noqa: F821
+        strategy: Literal["simple", "bootstrap"] = "simple",
+        confidence_level: float = 0.95,
+        n_resamples: int = 9999,
+        device: int = None,
+        random_state: Optional[int] = None,
+        input_column: str = "text",
+        label_column: str = "label",
+        generation_kwargs: dict = None,
```

Comment:
Clarification question (mostly for me!): these are limited to the kwargs passable to Text2TextGenerationPipeline, right? i.e. you wouldn't use this to pass in temperature or more granular sampler settings, for example

Reply:
Got it, ok thanks!